allennlp/models/decomposable_attention.py

```diff
@@ -164,13 +165,29 @@ def forward(self,  # type: ignore
 
         output_dict = {"label_logits": label_logits, "label_probs": label_probs}
 
+        self.add_insight_to_output_dict(metadata, output_dict, p2h_attention, h2p_attention)
+
         if label is not None:
             loss = self._loss(label_logits, label.long().view(-1))
             self._accuracy(label_logits, label.squeeze(-1))
             output_dict["loss"] = loss
 
         return output_dict
 
+    @staticmethod
+    def add_insight_to_output_dict(metadata, output_dict, p2h_attention, h2p_attention):
+        premise_tokens = []
+        hypothesis_tokens = []
+        if metadata is not None:
+            for datum in metadata:
+                premise_tokens.append(datum['premise_tokens'][:len(datum['premise_tokens'])-1])
```

Comment:
Just FYI (because this should be removed, anyway), the `MetadataField` does not get padded, so you don't need to do this `[:len...]` stuff.  It will just be the exact list you created in the `DatasetReader`.